lib/full-site-editing/block-templates.php

```diff
@@ -286,32 +290,36 @@ function _gutenberg_build_template_result_from_post( $post ) {
 /**
  * Retrieves a list of unified template objects based on a query.
  *
- * @param array $query {
+ * @param array  $query {
  *     Optional. Arguments to retrieve templates.
  *
  *     @type array  $slug__in List of slugs to include.
  *     @type int    $wp_id Post ID of customized template.
  * }
- * @param array $template_type wp_template or wp_template_part.
+ * @param string $template_type wp_template or wp_template_part.
+ * @param bool   $active        Whether to fetch active templates (default) or inactive.
  *
  * @return array Templates.
  */
-function gutenberg_get_block_templates( $query = array(), $template_type = 'wp_template' ) {
+function gutenberg_get_block_templates( $query = array(), $template_type = 'wp_template', $active = true ) {
+	// Temporarily disable inactive access for 5.8 version.
+	if ( ! $active ) {
+		return array();
+	}
+
+	$theme_slugs = get_theme_mod( $template_type, array() );
```

Comment:
This is a bit confusing -- it's an associative array that maps template _slugs_ (keys) to post _IDs_ (values); the same array is called `$ids` in [_gutenberg_build_template_result_from_post](https://github.com/WordPress/gutenberg/pull/31971/files#diff-6ca30c26729f0fb3398ad5c6b4e82d1a83bfe5bd920e4cb6deca5646d9e578bdR254).

I think `$ids` or `$theme_ids` might be the better name, as it reflects the array values, rather than the keys.

Reply:
Amended.